src/Pos/Launcher/Runner.hs

```diff
@@ -185,15 +193,38 @@ runRawRealMode res np@NodeParams {..} sscnp listeners outSpecs (ActionSpec actio
        runDBHolder modernDBs . runCH np initNC $ initNodeDBs
        initTip <- runDBHolder modernDBs getTip
        stateM <- liftIO SM.newIO
+       stateM_ <- liftIO SM.newIO
+
+       -- TODO need an effect-free way of running this into IO.
+       let runIO :: forall t . RawRealMode ssc t -> IO t
+           runIO = runProduction .
```

Comment:
About this: to set up EKG, we need to be able to run the special monad `m`, in this state `RawRealMode`, into `IO`, i.e. we need `forall t . RawRealMode t -> IO t`. BUT we need to do this in an 'effect-free' way. Unfortunately, some of the 'runner' functions, `runSscHolder` in particular, have extra side-effects. Ideally we would do the ssc runner side-effects once and use the resulting values to obtain `SscHolder m t -> m t` which can we re-used for this `runIO` and also for the function below which runs the node into `Production`.